benches/thrust/exclusive_scan/by_key.cu

```diff
@@ -0,0 +1,63 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/scan.h>
+
+#include <tbm/range_generator.cuh>
+
+template <typename T>
+thrust::device_vector<T> gen_keys(thrust::device_vector<T> keys)
```

Comment:
This will do unnecessary copying. It looks like the return value isn't used, consider just accepting a device vector reference and dropping the return type.